src/crypto/mod.rs

```diff
@@ -12,22 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use aes::cipher::{NewCipher, StreamCipher};
-use aes::Aes128Ctr;
-use digest::generic_array::{ArrayLength, GenericArray};
-use digest::{BlockInput, FixedOutput, Reset, Update};
-use hmac::{crypto_mac, Hmac, Mac, NewMac};
+use aes::{
+    cipher::{KeyIvInit, StreamCipher},
+    Aes128,
+};
+use digest::{
+    block_buffer::Eager,
+    consts::U256,
+    core_api::{BlockSizeUser, BufferKindUser, CoreProxy, FixedOutputCore},
+    generic_array::GenericArray,
+    typenum::{IsLess, Le, NonZero},
+    CtOutput, HashMarker,
+};
+use hmac::{Hmac, Mac};
 
-pub mod keys;
-
-// to not break existing imports
-pub use keys::*;
+//type export and aliasing to keep compatibility
+pub use x25519_dalek::PublicKey;
+pub type SharedSecret = x25519_dalek::PublicKey;
+pub type PrivateKey = x25519_dalek::StaticSecret;
+pub type EphemeralSecret = x25519_dalek::StaticSecret;
 
 pub const STREAM_CIPHER_KEY_SIZE: usize = 16;
 pub const STREAM_CIPHER_INIT_VECTOR: [u8; 16] = [0u8; 16];
 
-// Type alias for ease of use so that it would not require explicit import of crypto_mac or Hmac
-pub type HmacOutput<D> = crypto_mac::Output<Hmac<D>>;
+// Type alias for ease of use
+pub type HmacOutput<D> = CtOutput<Hmac<D>>;
+type Aes128Ctr = ctr::Ctr64BE<Aes128>;
```

Comment:
sanity check question: are we sure thats the correct endianness?

Reply:
Yes. aes crate `0.7.4` contains `pub type Aes128Ctr = ::ctr::Ctr64BE<Aes128>;` here : https://docs.rs/aes/0.7.4/src/aes/autodetect.rs.html#141